src/getting-started.md

```diff
@@ -0,0 +1,477 @@
+config, so this may result in different style from normal `rustfmt` if you have
+format-on-save turned on. It's a good habit to run `./x.py fmt` before every
+commit, as this reduces conflicts later. The pinned verson is built under
+`build/<target>/stage0/bin/rustfmt`, so if you want, you can use it for a
+single file or for format-on-save in your editor, which can be faster than `./x.py fmt`.
+
+On last thing: you can use `RUSTC_LOG=XXX` to get debug logging. [Read more
+here][logging]. Notice the `C` in `RUSTC_LOG`. Other than that, it uses normal
+[`env_logger`][envlog] syntax.
+
+[envlog]: https://crates.io/crates/env_logger
+[logging]: ./compiler-debugging.html#getting-logging-output
+
+### Building and Testing `std`/`core`/`alloc`/`test`/`proc_macro`/etc.
+
+As before, technically the proper way to build one of these libraries is to use
+the stage-2 compiler, which of course requires a 2-stage build, described above
+(`./x.py build`).
+
+In practice, though, you don't need to build the compiler unless you are
+planning to use a recently added nightly feature. Instead, you can just build
+stage 0 (i.e. which basically just uses the current beta compiler).
+
+```sh
+./x.py build --stage 0 src/libstd
+```
+
+```sh
+./x.py test --stage 0 src/libstd
+```
+
+(The same works for `src/liballoc`, `src/libcore`, etc.)
+
+### Building and Testing `rustdoc`
+
+`rustdoc` uses `rustc` internals (and, of course, the standard library), so you
+will have to build the compiler and `std` once before you can build `rustdoc`.
+As before, you can use `./x.py build` to do this.
+
+However, in practice, stage 1 should be sufficient. The first time you build,
+the stage-1 compiler will also be built.
+
+```sh
+# First build
+./x.py build --stage 1 src/tools/rustdoc
+
+# Subsequent builds
+./x.py build --stage 1 --keep-stage 1 src/tools/rustdoc
+```
+
+As with the compiler, you can do a fast check build:
+
+```sh
+./x.py check
+```
+
+Rustdoc has two types of tests: content tests and UI tests.
+
+```sh
+# Content tests
+./x.py test --stage 1 src/test/rustdoc
+
+# UI tests
+./x.py test --stage 1 src/test/rustdoc-ui
+
+# Both at once
+./x.py test --stage 1 src/test/rustdoc*
+```
+
+### Contributing code to other Rust projects
+
+There are a bunch of other projects that you can contribute to outside of the
+`rust-lang/rust` repo, including `clippy`, `miri`, `chalk`, and many others.
+
+These repos might have their own contributing guidelines and procedures. Many
+of them are owned by working groups (e.g. `chalk` is largely owned by
+WG-traits). For more info, see the documentation in those repos' READMEs.
+
+### Other ways to contribute
+
+There are a bunch of other ways you can contribute, especially if you don't
+feel comfortable jumping straight into the large `rust-lang/rust` codebase.
+
+The following tasks are doable without much background knowledge but are
+incredibly helpful:
+
+- [ICE-breakers Cleanup crew][iceb]: find minimal reproductions of ICEs, bisect
```

Comment:
I think this isn't called "ICE-breaker" anymore. I think the new terminology is simply "Cleanup crew" or "Cleanup crew notification group".
```suggestion
- [ICleanup crew][iceb]: find minimal reproductions of ICEs, bisect
```